### ISSUE TITLE
Update cmake files to build on macOS

### ISSUE THREAD

**Issue by jonkeane** [libhand/libhand#5]
After installing dependencies, it is possible to build on macOS 10.12.

Currently ogre plugins do not link correctly, but it works (with modifications to the cmake of hand_cpp).

Need to:
- [x] adjust cmake (Ideally, this wouldn't mean linking directly to the homebrew path)
- [x] fix problems with ogre linking
- [ ] investigate if ogre 1.9 can be used.

**Comment by shasheene** [comment]
Assuming I'm thinking of the same issue you're having, I'm very familiar with this issue (on Debian and Windows in particular). I don't think I've documented some of my investigation into OGRE linking/loading other than some old commit messages, so I'll do a bit of a braindump here (sorry for the verbosity but some of it may or may not be useful).

If we're talking about an issue where OGRE fails to load its plugins at runtime, the 'true' solution is to supply OGRE runtime with a plugins.cfg file pointing to the fully qualified paths of the OGRE plugins. However, the paths are platform and cpu architecture dependent, and the upstream OGRE developers [do NOT ship with a default plugins.cfg](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=826384) for us to use.

By the way, OGRE has different API calls to loading plugins depending on if the plugins were statically or dynamically linked. In my mind, libhand should support both types of linking depending on user preference. And in my mind the best way of specifying how OGRE should be linked is with a boolean in CMakeLists.txt file, which led me to add some relatively ugly #ifdef's in /hand_cpp/source/hand_renderer.cc

That solution worked well enough to support both dynamic and static linking OGRE with Debian/Ubuntu. On Windows, the plugin filenames were slightly different (ugh). On Linux, when loading is dynamic, the dynamic linker/loader (`man ld.so`) searches the /etc/ld.so.conf (`man ldconfig`). OGRE doesn't add to /etc/ld.so/conf (see Debian mailing list above), so the instructions at /docs/build_instructions/debian/BUILD.DEBIAN.md ask the user to append to LD_LIBRARY_PATH. A ugly stop gap solution which I don't like, but it was the most efficient way to maintain backwards compatibility with v0.9.

I will mention moving to the plugins.cfg solution is the 'correct' solution according to the OGRE developers. But this would mean supplying a separate plugins.cfg for each platform (and CPU architecture) and would break backwards compatibility with LibHand 0.9 (which I'm extremely hesitant to do yet). I actually have a branch where I added a shell script which searched a preset bunch of directories and generate the correct plugins.cfg per platform and was considering porting it to C++ to run at runtime.

As far as this particular issue goes, I suspect given it builds (and therefore successfully found the CMake headers etc) and fails on startup it's likely that it can't find the plugin it wants to load (eg, Octtree

Unfortunately there is unlikely a solution which works on both Linux, Windows and macOS and maintains backwards compatibility with existing applications built against Libhand v0.9.0. Which will probably lead to another #ifdef for now (assuming this issue is the one I'm thinking of).

In my previous work on this code base, I've come to the conclusion that for LibHand's future development it needs to move away from OGRE 3D for various reasons. That of course would be an API breaking change so will get a new major version number (bundled in with other major improvements).

**Comment by jonkeane** [comment]
I've (finally!) had some time to look at this, and I've managed to get it working. I've created #7 which has the changes to build libhand on macOS.

Thank you very much for pointing the plugin issue out to me, that was exactly what was going on. I did have to add some ugly `#ifdef`s into `/hand_cpp/source/hand_renderer.cc`. For versions of macOS / OS X with SIP, it appears that using the `DYLD_FRAMEWORK_PATH` environment variable is no longer supported (cf https://groups.google.com/forum/#!topic/ruby-ffi/1J8apii7-0g and https://bugs.openjdk.java.net/browse/JDK-8139288). For this reason, I've included full paths to the ogre libraries, which I believe are somewhat stable if installed via homebrew. 

I totally agree that this solution is not great long-term, but in the interest of maintaining backwards compatibility does work for now.
